test/method/testvolume.py

```diff
@@ -57,9 +57,35 @@ def test_density(self):
 
         vol = volume.Volume((-3.0, -6.0, -2.0), (3.0, 6.0, 2.0), (0.25, 0.25, 0.25))
 
-        frontierorbs = [data_sp.mocoeffs[0][(data_sp.homos[0] - 3):(data_sp.homos[0] + 1)]]
+        frontierorbs = [data_sp.mocoeffs[0][(data_sp.homos[0] - 3) : (data_sp.homos[0] + 1)]]
         density = volume.electrondensity(data_sp, vol, frontierorbs)
         integral = density.integrate()
 
         self.assertTrue(abs(integral - 8.00) < 1e-2)
         print("Combined Density of 4 Frontier orbitals=", integral)
+
+    def test_cube(self):
+        """Does the cube file written out for electron density on a Cartesian grid match
+        expected values?
+        """
+
+        data, logfile = getdatafile(Psi4, "basicPsi4-1.2.1", ["water_mp2.out"])
+
+        # Reference values were calculated using cubegen method in Psi4.
+        # First six rows are information about the coordinates of the grid and comments.
+        tmp = numpy.array([])
+        with open(os.path.dirname(os.path.realpath(__file__)) + "/water_mp2.cube") as f:
+            lines = f.readlines()
+            for line in lines[6 : len(lines)]:
+                tmp = numpy.append(tmp, numpy.fromstring(line, dtype=float, sep=" "))
```

Comment:
I think you can avoid the loop with append by using `np.fromtxt` and skipping the first 6 rows.

Reply:
Yes. I'll try using `np.fromtxt` with `skiprows = 6` after the call and update this. Thanks for the comment!

Reply:
@berquist I've tried using `np.loadtxt` and it looks like `loadtxt` expects every row to have same number of elements. This is not the case for cube files generated by Psi4. I can still use `loadtxt` for the first N-1 lines then do `fromstring` for the last line -- I'm not sure which would be better though.

Reply:
In that case, I would read the file as a list of lists of strings, and then have a single call to `np.asanyarray` where the dtype is set to floats, and then call the `resize` method on that.

Reply:
I have just updated this to read in as strings first and then construct numpy array in a single call. Thanks.